taichi/ir/snode.cpp

```diff
@@ -207,11 +207,12 @@ std::string SNode::get_node_type_name_hinted() const {
   return fmt::format("S{}{}{}", id, snode_type_name(type), suffix);
 }
 
-int SNode::get_num_bits(int physical_index) const {
+int SNode::get_num_bits(int virtual_index) const {
   int result = 0;
   const SNode *snode = this;
+  auto physical_index = snode->physical_index_position[virtual_index];
   while (snode) {
-    result += extractors[physical_index].num_bits;
+    result += snode->extractors[physical_index].num_bits;
     snode = snode->parent;
   }
```

Comment:
There are two issues here:
 - We should use virtual indices as input and do a translation within the function (see the doc updates)
 - `snode->extractors` instead of `extractors` :-)

Reply:
For the second point, sorry for writing a bug here :)

For the first point, I think we're already using physical indices in `LoopIndexStmt`s. See
https://github.com/taichi-dev/taichi/blob/ca81fdff67d05caa1d41f21b13ae00c46ef6485c/taichi/transforms/lower_ast.cpp#L286-L287

Reply:
Ah you are right! I just now reverted that modification.